ui/app/components/clients/activity.ts

```diff
@@ -0,0 +1,178 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+// base component for counts child routes that can be extended as needed
+// contains getters that filter and extract data from activity model for use in charts
+
+import Component from '@glimmer/component';
+import { isAfter, isBefore, isSameMonth, fromUnixTime } from 'date-fns';
+import { parseAPITimestamp } from 'core/utils/date-formatters';
+
+import type ClientsActivityModel from 'vault/models/clients/activity';
+import type ClientsVersionHistoryModel from 'vault/models/clients/version-history';
+
+interface Args {
```

Comment:
Are these the args that have to be passed into any component that extends from `ClientsActivityComponent`

Reply:
Yes they are needed by this component class. The extending component could have additional args if need be.